src/main/java/Parser.java

```diff
@@ -0,0 +1,80 @@
+public class Parser {
+
+    public Parser() {
+    }
+
+    public static Command parse(String command) throws DukeException {
+        String[] commandSplit = command.split(" ");
+        String type = commandSplit[0];
+
+        Type enumType;
+        String timeDesc;
+
+        switch (type) {
+            case "list":
+                return new ListCommand();
+
+            case "done":
+                if (commandSplit.length == 1) {
+                    throw new DukeException("☹ OOPS!!! Please indicate which task you have completed.");
+                }
+
+                return new DoneCommand(commandSplit[1]);
+
+            case "delete":
+                if (commandSplit.length == 1) {
+                    throw new DukeException("☹ OOPS!!! Please indicate which task you would like to delete.");
+                }
+
+                return new DeleteCommand(commandSplit[1]);
+
+            case "bye":
+                return new ExitCommand();
+
+            case "todo":
+            case "deadline":
+            case "event":
```

Comment:
Nice job using Java's switch statement for this method. I think it does make things a little more readable as compared to if else blocks. However, I feel that your choice of using nested switch here does backfire a bit in terms of readability, maybe you can consider listing our the case for "todo", "deadline" and "event" directly? Just my two cents!